AGENTS.md misdescribes the test fixtures: no window/pane fixtures exist, and Server in doctests is a TestServer partial

## Bug

`AGENTS.md` describes the fixtures a contributor may use, and gets three things wrong. Each one sends someone down a path that does not work — and the guidance is aimed at agents and newcomers, i.e. exactly the readers least able to tell it is wrong.

## 1. There are no `window` or `pane` fixtures for tests

[`AGENTS.md#L188`](https://github.com/tmux-python/libtmux/blob/v0.61.0/AGENTS.md?plain=1#L188):

> - For libtmux, use provided fixtures: `server`, `session`, `window`, and `pane`

The pytest plugin exports only **`server`** and **`session`** ([`src/libtmux/pytest_plugin.py#L145`](https://github.com/tmux-python/libtmux/blob/v0.61.0/src/libtmux/pytest_plugin.py#L145), [`#L224`](https://github.com/tmux-python/libtmux/blob/v0.61.0/src/libtmux/pytest_plugin.py#L224)). Its full fixture set is `home_path`, `home_user_name`, `user_path`, `zshrc`, `config_file`, `clear_env`, `server`, `session_params`, `session`, `control_mode` — no `window`, no `pane`.

`window` and `pane` exist **only in the doctest namespace** ([`conftest.py#L51-L52`](https://github.com/tmux-python/libtmux/blob/v0.61.0/conftest.py#L51-L52)), which is a different mechanism. A test written from `AGENTS.md` fails at collection:

```
def test_something(pane: Pane) -> None:
E       fixture 'pane' not found
>       available fixtures: ..., server, session, ...
```

The existing tests already work around this by deriving the pane themselves (`window = session.new_window(...)`, `pane = window.active_pane`) — the file just never says so.

## 2. `Server` in the doctest namespace is not the `Server` class

[`AGENTS.md#L254`](https://github.com/tmux-python/libtmux/blob/v0.61.0/AGENTS.md?plain=1#L254) lists `Server` among the doctest names, implying it is the class. `conftest.py` binds the real class and then **overwrites it**:

```python
doctest_namespace["Server"] = Server                                # conftest.py:42
...
doctest_namespace["Server"] = request.getfixturevalue("TestServer") # conftest.py:48  <- wins
```

So in a doctest, `Server` is a `functools.partial` (a `TestServer` factory), not `libtmux.Server`. Any doctest calling a **classmethod** — `Server.from_env(...)`, and by extension anything added to that family — dies with:

```
AttributeError: 'functools.partial' object has no attribute 'from_env'
```

The workaround is `>>> from libtmux.server import Server`, which `src/libtmux/pytest_plugin.py` already does in its own docstrings — but `AGENTS.md` never mentions the trap.

## 3. The doctest-namespace list is incomplete

`AGENTS.md#L254` lists: `server`, `session`, `window`, `pane`, `Server`, `Session`, `Window`, `Pane`, `request`.

`conftest.py` also seeds **`Client`** (`:46`), **`ControlMode`** (`:54`), **`control_mode`** (`:55`) and **`monkeypatch`** (`:60`). `monkeypatch` in particular is genuinely useful — it is what lets a doctest set `TMUX` / `TMUX_PANE` and demonstrate an environment-reading API for real, instead of reaching for `# doctest: +SKIP` (which the file rightly bans).

## Suggested fix

In `AGENTS.md`:

- Correct the test-fixture list to `server` and `session`, and say that a window or pane is derived from the session (`session.new_window(...)` / `window.active_pane`) — with a pointer to `src/libtmux/pytest_plugin.py` as the source of truth.
- Correct the doctest-namespace list to what `conftest.py` actually seeds, including `Client`, `ControlMode`, `control_mode` and `monkeypatch`.
- Warn that `Server` in a doctest is a `TestServer` partial, and that `>>> from libtmux.server import Server` is the way to reach the class.

Worth adding while there: the fixture server is created with a socket *name*, so `server.socket_path` is `None` — a doctest that needs the path has to ask tmux for it (`display-message -p '#{socket_path}'`). That has bitten at least one doctest already.

## Related

- `docs/AGENTS.md` carried a separate false claim about doctest blocks sharing one session; that one is corrected in #705. This issue is the root `AGENTS.md`, which is untouched by it.

## Environment

- libtmux: v0.61.0 (`AGENTS.md`, `conftest.py`, `src/libtmux/pytest_plugin.py` all unchanged since the tag)

--- a/AGENTS.md
+++ b/AGENTS.md
@@ -166,26 +166,30 @@ Server (tmux server instance)
 
 8. **pytest Plugin** (`src/libtmux/pytest_plugin.py`)
    - Provides fixtures for testing with tmux
-   - Creates temporary tmux sessions/windows/panes
+   - Creates temporary tmux servers and sessions
 
 ## Testing Strategy
 
-libtmux uses pytest for testing with custom fixtures. The pytest plugin (`pytest_plugin.py`) defines fixtures for creating temporary tmux objects for testing. These include:
+libtmux uses pytest for testing with custom fixtures. The pytest plugin
+(`src/libtmux/pytest_plugin.py`) is the source of truth for the fixtures it
+provides. Its ready-to-use tmux hierarchy object fixtures are:
 
 - `server`: A tmux server instance for testing
 - `session`: A tmux session for testing
-- `window`: A tmux window for testing
-- `pane`: A tmux pane for testing
 
-These fixtures handle setup and teardown automatically, creating isolated test environments.
+These fixtures handle setup and teardown automatically, creating isolated test
+environments. For regular tests, derive windows and panes from the `session`
+fixture with `session.new_window(...)` and `window.active_pane`. The `window`
+and `pane` names described below are doctest namespace values, not pytest
+fixtures.
 
 ### Testing Guidelines
 
 1. **Use functional tests only**: Write tests as standalone functions, not classes. Avoid `class TestFoo:` groupings - use descriptive function names and file organization instead.
 
 2. **Use existing fixtures over mocks**
-   - Use fixtures from conftest.py instead of `monkeypatch` and `MagicMock` when available
-   - For libtmux, use provided fixtures: `server`, `session`, `window`, and `pane`
+   - Use fixtures from `src/libtmux/pytest_plugin.py` and `conftest.py` instead of `monkeypatch` and `MagicMock` when available
+   - For regular libtmux tests, use the `server` and `session` object fixtures; derive windows and panes from the session
    - Document in test docstrings why standard fixtures weren't used for exceptional cases
 
 3. **Preferred pytest patterns**
@@ -199,11 +203,16 @@ These fixtures handle setup and teardown automatically, creating isolated test e
 ### Example Fixture Usage
 
 ```python
-def test_window_rename(window):
-    """Test renaming a window."""
-    # window is already a Window instance with a live tmux window
-    window.rename_window('new_name')
-    assert window.window_name == 'new_name'
+def test_window_rename(session):
+    """Test renaming a window derived from the session fixture."""
+    window = session.new_window(window_name="old_name")
+    pane = window.active_pane
+    assert pane is not None
+
+    window.rename_window("new_name")
+
+    assert window.window_name == "new_name"
+    assert pane.window_id == window.window_id
 ```
 
 ## Coding Standards
@@ -251,9 +260,24 @@ type
 - If you cannot create a working doctest, **STOP and ask for help**
 
 **Available tools for doctests:**
-- `doctest_namespace` fixtures: `server`, `session`, `window`, `pane`, `Server`, `Session`, `Window`, `Pane`, `request`
+- `doctest_namespace` values: `server`, `session`, `window`, `pane`, `Server`, `Session`, `Window`, `Pane`, `Client`, `request`, `ControlMode`, `control_mode`, `monkeypatch`
 - Ellipsis for variable output: `# doctest: +ELLIPSIS`
 - Update `conftest.py` to add new fixtures to `doctest_namespace`
+
+In doctests, `Server` is the `TestServer` factory from
+`src/libtmux/pytest_plugin.py`, a `functools.partial` that creates uniquely
+named servers and registers their cleanup. It is not the
+`libtmux.server.Server` class. Import the real class explicitly when a doctest
+needs class-level APIs:
+
+```python
+>>> from libtmux.server import Server
+```
+
+The `server` fixture and the doctest `Server()` factory construct servers with
+socket names, so their `socket_path` attributes are `None`. When a doctest
+needs the actual path, query tmux's `#{socket_path}` format with
+`server.cmd("display-message", "-p", "#{socket_path}").stdout[0]`.
 
 **`# doctest: +SKIP` is NOT permitted** - it's just another workaround that doesn't test anything. Use the fixtures properly - tmux is required to run tests anyway.
 

